From a4be1b5dd42798d1cc0e26ee5fed09054715c5b2 Mon Sep 17 00:00:00 2001
From: Christoph Gerum <christoph@gerum.de>
Date: Tue, 26 Oct 2010 13:29:31 +0200
Subject: [PATCH] Don't check for same operands in mul and mla.

---
 machine.def    |   19 ++-----------------
 sim-outorder.c |    3 ++-
 2 files changed, 4 insertions(+), 18 deletions(-)

diff --git a/machine.def b/machine.def
index 64c7717..f52c8b9 100644
--- a/machine.def
+++ b/machine.def
@@ -1492,23 +1492,14 @@ DEFINST(MVNSI,			0x0f,
 	DGPR(RD), DPSR, DNA,	DCOND, DNA, DNA, DNA)
 
 
-
-
-
-
-     /* ALL OF THE MULL INSTRUCTIONS */
+					       
+/* ALL OF THE MULL INSTRUCTIONS */
 CONNECT(MUL_LINK)
 
 #define MUL_IMPL							\
   {									\
     if (COND_VALID(PSR))						\
       {									\
-	if ((RM) == (RN))						\
-	  {								\
-	    SET_GPR(RN, 0);						\
-	    /* FIXME: should this be "md_fault_dsteqop1"? */		\
-	    DECLARE_FAULT(md_fault_invalid);				\
-	  }								\
 	if ((RN) == MD_REG_PC)						\
 	  {								\
 	    /* FIXME: should this be "md_fault_pcdst"? */		\
@@ -1558,12 +1549,6 @@ DEFINST(MULS,			0x01,
   {									\
     if (COND_VALID(PSR))						\
       {									\
-	if ((RM) == (RN))						\
-	  {								\
-	    SET_GPR(RN, GPR(RD));					\
-	    /* FIXME: should this be "md_fault_dsteqop1"? */		\
-	    DECLARE_FAULT(md_fault_invalid);				\
-	  }								\
 	if ((RN) == MD_REG_PC)						\
 	  {								\
 	    /* FIXME: should this be "md_fault_pcdst"? */		\
diff --git a/sim-outorder.c b/sim-outorder.c
index 07174b0..7550940 100644
--- a/sim-outorder.c
+++ b/sim-outorder.c
@@ -4271,9 +4271,10 @@ ruu_dispatch(void)
 
 	  /*PATCH bx instructions*/
 	  if(((inst >> 16)  & 0xfff) == 0x12f){
-	    //printf("bx inst: %08x, pc: %x\n", inst, regs.regs_PC);
+	    
 	    int rm = inst & 0xf;
 	    int cond = inst &  0xf0000000;
+	    //printf("bx inst: %08x, pc: %x, cond: %x, rm: %x \n", inst, regs.regs_PC, cond, rm);
 	    inst = cond | rm | 0x01a0f000;
 	  }
 
-- 
1.6.0.4

